### PR TITLE
doc: improve module imports and tweak module docs

### PR DESCRIPTION
**Pull request by ntrel**
~~Rename 'Modules' heading -> 'Modules 2' (which explains how to create named modules). This follows the pattern of Functions, Functions 2; Types, Types 2 in the table of contents.
Group 'Module imports' under new heading 'Modules' (the basics),~~
~~Move 'Symbol visibility' to subsection of 'Modules'.~~

Add link to standard library docs.
Explain main default module, all files in folder form a module.
Move paragraph about module prefixes to 'Module imports' section (and remove now unnecessary see also link).
Add heading for 'Selective imports' & add example for importing multiple specific symbols.

Rendered: https://github.com/ntrel/V/blob/doc-mod/doc/docs.md#table-of-contents

### CODE REVIEW

File: doc/docs.md [hunk 1]
```diff
@@ -604,33 +591,73 @@ numbers := {
...
+```
+
+Functions are private (not exported) by default.
+To allow other modules to use them, prepend `pub`. The same applies
+to constants and types. 
 
-For information about creating a module, see [Modules](#modules)
+Note: `pub` can only be used from a named module.
 
-### Importing a module
```
Comment (ntrel) [inline doc/docs.md:611]: I think the `Module imports` anchor link would be used instead of this, but if you like I can manually add an anchor for this.

File: doc/docs.md [hunk 2]
```diff
@@ -604,33 +591,73 @@ numbers := {
...
+```v
+import os { input, user_os }
+
+name := input('Enter your name: ')
+println('Name: $name')
+os := user_os()
+println('Your OS is ${os}.')
+```
+
+#### Module import aliasing
```
Comment (ntrel) [inline doc/docs.md:660]: Might be worth moving this subsection in another PR (e.g. under Modules 2) because it is more advanced.

Comment (JalonSolov) [reply doc/docs.md:660]: I think that should be in the regular `Importing a module` section.  It's better to keep the information for a subject close together, to avoid having to search all over the place to find pieces.  If you want to keep them separate, then the first section should have a link to the 2nd, or people might think the first section is all there is on that subject.

**Comment by ntrel** [comment]
I decided to remove all the changes to module sections and grouping from this pull, the diff should be much easier to follow now.
